### PR TITLE
hub: Correctly handle empty `DEVEL_EMAIL_ADDRESS`

### PR DESCRIPTION
**Pull request by siteshwar**

### CODE REVIEW

File: osh/hub/scan/notify.py [hunk 1]
```diff
@@ -264,9 +264,13 @@ def send_scan_notification(request, scan_id):
     # recipient setting
     recipient = get_recipient(scan.username)
 
-    # message setting
+    recipients = set()
```
Comment (kdudka) [inline osh/hub/scan/notify.py:267]: I do not like the identifiers differing from each other with the trailing `s` only.  Could we please use `recipient_list` or `recipient_set` for the new one?

**Review by kdudka — APPROVED** [review]

File: osh/hub/scan/notify.py [hunk 2]
```diff
@@ -291,7 +295,7 @@ def send_scan_notification(request, scan_id):
         "X-Scan-Package": scan.package.name,
         "X-Scan-Build": scan.nvr,
     }
-    return send_mail(message, subject, [recipient], headers)
+    return send_mail(message, subject, recipients, headers)
```
Comment (hanchuntao) [inline osh/hub/scan/notify.py:298]: recipients is a `list` or an `str`, why need to change it to a `set`

Comment (siteshwar) [reply osh/hub/scan/notify.py]: I copied the code from a [similar place](https://github.com/openscanhub/openscanhub/blob/6e1fb9f774fde5bbf094d0ac9ac6b3c71b1fb1e7/osh/hub/scan/notify.py#L144). I would try to fix it now.

File: osh/hub/scan/notify.py [hunk 3]
```diff
@@ -293,7 +297,7 @@ def send_scan_notification(request, scan_id):
         "X-Scan-Package": scan.package.name,
         "X-Scan-Build": scan.nvr,
     }
-    return send_mail(message, subject, [recipient], headers)
+    return send_mail(message, subject, recipient_list, headers)
```
[the remark this answers is not in the document]
Comment (siteshwar) [reply osh/hub/scan/notify.py:300]: I already converted it to a `list`. Do we need a `set` variable?

[the remark this answers is not in the document]
Comment (lzaoral) [reply osh/hub/scan/notify.py:300]: What will happen if the list contains some recipient twice?  Either way, set is a nice way to get rid of such duplicates.

[the remark this answers is not in the document]
Comment (siteshwar) [reply osh/hub/scan/notify.py:300]: The [EmailMessage](https://docs.djangoproject.com/en/5.0/topics/email/#django.core.mail.EmailMessage) class takes `to` argument as a list. So it is not mandatory to use set here.

[the remark this answers is not in the document]
Comment (lzaoral) [reply osh/hub/scan/notify.py:300]: Let it be a list then.  My argument that the indentation is excessive and unnecessary still stands.

You can use `recipient_list = [r for r in recipient_list if r]` to filter out empty strings at the end.

File: osh/hub/scan/notify.py [hunk 4]
```diff
@@ -266,9 +266,12 @@ def send_scan_notification(request, scan_id):
     # recipient setting
     recipient = get_recipient(scan.username)
 
-    # message setting
+    recipient_list = list()
+
+    recipient_list.append(recipient)
```
Comment (lzaoral) [inline osh/hub/scan/notify.py:271]: This whole section can be simplified like this:
```python
recipient_list = [get_recipient(scan.username)]
```

**Review by lzaoral — APPROVED** [review]

**Pull request merged**